cmd/kubectl/kubectlcobra/applier.go

```diff
@@ -0,0 +1,200 @@
+// performs prune to clean up any resources that has been deleted.
+type Applier struct {
+	factory   util.Factory
+	ioStreams genericclioptions.IOStreams
+
+	applyOptions  *apply.ApplyOptions
+	statusOptions *StatusOptions
+	resolver      resolver
+}
+
+// Initialize sets up the Applier for actually doing an apply against
+// a cluster. This involves validating command line inputs and configuring
+// clients for communicating with the cluster.
+func (a *Applier) Initialize(cmd *cobra.Command) error {
+	a.applyOptions.PreProcessorFn = PrependGroupingObject(a.applyOptions)
+	err := a.applyOptions.Complete(a.factory, cmd)
+	if err != nil {
+		return errors.WrapPrefix(err, "error setting up ApplyOptions", 1)
+	}
+
+	resolver, err := a.newResolver(a.statusOptions.period)
+	if err != nil {
+		return errors.WrapPrefix(err, "error creating resolver", 1)
+	}
+	a.resolver = resolver
+	return nil
+}
+
+// SetFlags configures the command line flags needed for apply and
+// status. This is a temporary solution as we should separate the configuration
+// of cobra flags from the Applier.
+func (a *Applier) SetFlags(cmd *cobra.Command) {
+	a.applyOptions.DeleteFlags.AddFlags(cmd)
+	a.applyOptions.RecordFlags.AddFlags(cmd)
+	a.applyOptions.PrintFlags.AddFlags(cmd)
+	a.statusOptions.AddFlags(cmd)
+	a.applyOptions.Overwrite = true
+}
+
+// newResolver sets up a new Resolver for computing status. The configuration
+// needed for the resolver is taken from the Factory.
+func (a *Applier) newResolver(pollInterval time.Duration) (*wait.Resolver, error) {
+	config, err := a.factory.ToRESTConfig()
+	if err != nil {
+		return nil, errors.WrapPrefix(err, "error getting RESTConfig", 1)
+	}
+
+	mapper, err := a.factory.ToRESTMapper()
+	if err != nil {
+		return nil, errors.WrapPrefix(err, "error getting RESTMapper", 1)
+	}
+
+	c, err := client.New(config, client.Options{Scheme: scheme.Scheme, Mapper: mapper})
+	if err != nil {
+		return nil, errors.WrapPrefix(err, "error creating client", 1)
+	}
+
+	return wait.NewResolver(c, mapper, pollInterval), nil
+}
+
+// Run performs the Apply step. This happens asynchronously with updates
+// on progress and any errors are reported back on the event channel.
+// Cancelling the operation or setting timeout on how long to wait
+// for it complete can be done with the passed in context.
+// Note: There sn't currently any way to interrupt the operation
+// before all the given resources have been applied to the cluster. Any
+// cancellation or timeout will only affect how long we wait for the
+// resources to become current.
+func (a *Applier) Run(ctx context.Context) <-chan Event {
+	ch := make(chan Event)
+
+	go func() {
+		defer close(ch)
+		adapter := &KubectlPrinterAdapter{
+			ch: ch,
+		}
+		// The adapter is used to intercept what is meant to be printing
+		// in the ApplyOptions, and instead turn those into events.
+		a.applyOptions.ToPrinter = adapter.toPrinterFunc()
```

Comment:
Here we end up losing a lot of current `kubectl` printing functionality with the different printers (through `PrintFlags`: e.g. `-o yaml`). Are we going to have rewrite those?